server/src/main/java/org/elasticsearch/repositories/RepositoryData.java

```diff
@@ -151,7 +180,14 @@ public RepositoryData withVersions(Map<SnapshotId, Version> versions) {
         final Map<String, Version> newVersions = new HashMap<>(snapshotVersions);
         versions.forEach((id, version) -> newVersions.put(id.getUUID(), version));
         return new RepositoryData(
-            genId, snapshotIds, snapshotStates, newVersions, indexSnapshots, shardGenerations, indexMetaDataGenerations);
+                genId,
+                snapshotIds,
+                snapshotStates,
+                newVersions,
+                indices,
```

Comment:
Adding `indices` so that this uses the private constructor to avoid re-wrapping everything in another `Collections.unmodifiableMap()`.